presto-main/src/main/java/com/facebook/presto/operator/scalar/ArraySortFunction.java

```diff
@@ -115,6 +120,134 @@ public int compare(Integer p1, Integer p2)
         return blockBuilder.build();
     }
 
+    @SqlType("array(bigint)")
+    public static Block bigintSort(@SqlType("array(bigint)") Block array)
+    {
+        final int arrayLength = array.getPositionCount();
+        if (arrayLength < 2) {
+            return array;
+        }
+
+        long[] values = new long[array.getPositionCount()];
+        int nulls = 0;
+
+        if (array.mayHaveNull()) {
+            int j = 0;
+            for (int i = 0; i < array.getPositionCount(); i++) {
+                if (array.isNull(i)) {
+                    nulls++;
+                }
+                else {
+                    values[j++] = BIGINT.getLong(array, i);
+                }
+            }
+        }
+        else {
+            for (int i = 0; i < array.getPositionCount(); i++) {
+                values[i] = BIGINT.getLong(array, i);
+            }
+        }
+
+        Arrays.sort(values, 0, values.length - nulls);
+
+        BlockBuilder blockBuilder = BIGINT.createBlockBuilder(null, arrayLength);
```

Comment:
Is it possible to not create extra array values, and reuse the values array in block builder?

Reply:
> Is it possible to not create extra array values, and reuse the values array in block builder?

Not really. That's internal and can vary so we want to follow the Block API properly. This is just an opportunistic specialization for primitive types.